From 51a601d63019bb807fce15c1346932c5162fb2db Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@cloudera.com>
Date: Mon, 28 Jan 2013 09:58:58 -0800
Subject: [PATCH 022/596] MR1: CLOUDERA-BUILD. ReduceTask class requires
 public visibility for pluggable sort.

  Reason: bug
  Author: Alejandro Abdelnur
  Ref: CDH-10154
(cherry picked from commit cd09304604d3ebaa850b8e78b3c8d186fe6e25e3)

(cherry picked from commit a33a846e220fadefcb7388664b3bd5a854d8d3cc)
(cherry picked from commit 3213daff8913ade29b727e23be92e8cf7f23d730)
(cherry picked from commit d824bfe407d593a69af2a919bd541509b91fd547)
---
 .../org/apache/hadoop/mapred/ReduceTask.java       |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
index 248acf7..3499cc2 100644
--- a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
+++ b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
@@ -53,6 +53,8 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.classification.InterfaceStability;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.ChecksumFileSystem;
 import org.apache.hadoop.fs.FSError;
@@ -92,7 +94,9 @@
 import org.apache.hadoop.mapreduce.security.SecureShuffleUtils;
 
 /** A Reduce task. */
-class ReduceTask extends Task {
+@InterfaceAudience.Private
+@InterfaceStability.Unstable
+public class ReduceTask extends Task {
 
   static {                                        // register a ctor
     WritableFactories.setFactory
-- 
1.7.9.5

